File: java/src/test/java/problems301_400/OddEvenLinkedListTest.java
package problems301_400;

import concepts.SinglyListNode;
import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class OddEvenLinkedListTest extends SampledTest {
    @Test
    public void test() {
        for (OddEvenLinkedList solution : OddEvenLinkedList.values()) {
            for (OddEvenLinkedListSample sample : getSamples(OddEvenLinkedListSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.oddEvenList(SinglyListNode.of(sample.input.head)))
                    .isEqualTo(SinglyListNode.of(sample.output));
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems301_400/OddEvenLinkedListTest.kt
package problems301_400

import com.google.common.truth.Truth.assertWithMessage
import concepts.SinglyListNode
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class OddEvenLinkedListTest : SampledTest() {
    @Test
    fun test() =
        OddEvenLinkedList.entries.forEach { solution ->
            getSamples<Array<OddEvenLinkedListSample>>().forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(solution.oddEvenList(SinglyListNode.of(*input.head)))
                    .isEqualTo(SinglyListNode.of(*output))
            }
        }
}
